sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyAsyncClient.java

```diff
@@ -529,50 +522,50 @@
      *     string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Key> getKey(KeyBase keyBase) {
-        return getKeyWithResponse(keyBase).flatMap(FluxUtil::toMono);
+    public Mono<Key> getKey(KeyProperties keyProperties) {
+        return getKeyWithResponse(keyProperties).flatMap(FluxUtil::toMono);
     }
 
     /**
-     * Get public part of the key which represents {@link KeyBase keyBase} from the key vault. The get key operation is
+     * Get public part of the key which represents {@link KeyProperties keyProperties} from the key vault. The get key operation is
      * applicable to all key types and it requires the {@code keys/get} permission.
      *
      * <p>The list operations {@link KeyAsyncClient#listKeys()} and {@link KeyAsyncClient#listKeyVersions(String)}
-     * return the {@link Flux} containing {@link KeyBase base key} as output excluding the key material of the key.
-     * This operation can then be used to get the full key with its key material from {@code keyBase}.</p>
+     * return the {@link Flux} containing {@link KeyProperties key properties} as output excluding the key material of the key.
+     * This operation can then be used to get the full key with its key material from {@code keyProperties}.</p>
      *
-     * {@codesnippet com.azure.security.keyvault.keys.async.keyclient.getKeyWithResponse#KeyBase}
+     * {@codesnippet com.azure.security.keyvault.keys.async.keyclient.getKeyWithResponse#KeyProperties}
      *
-     * @param keyBase The {@link KeyBase base key} holding attributes of the key being requested.
+     * @param keyProperties The {@link KeyProperties key properties} holding attributes of the key being requested.
      * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains the requested
      *     {@link Key key}.
-     * @throws ResourceNotFoundException when a key with {@link KeyBase#name() name} and {@link KeyBase#version()
+     * @throws ResourceNotFoundException when a key with {@link KeyProperties#getName() name} and {@link KeyProperties#getVersion()
      *     version} doesn't exist in the key vault.
-     * @throws HttpRequestException if {@link KeyBase#name()}  name} or {@link KeyBase#version() version} is empty
+     * @throws HttpRequestException if {@link KeyProperties#getName()}  name} or {@link KeyProperties#getVersion() version} is empty
      *     string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<Key>> getKeyWithResponse(KeyBase keyBase) {
-        Objects.requireNonNull(keyBase, "The Key Base parameter cannot be null.");
-        return withContext(context -> getKeyWithResponse(keyBase.name(), keyBase.version() == null ? ""
-            : keyBase.version(), context));
+    public Mono<Response<Key>> getKeyWithResponse(KeyProperties keyProperties) {
+        Objects.requireNonNull(keyProperties, "The Key Base parameter cannot be null.");
```

Comment:
The ~~Key Base~~ Key Properties parameter cannot be null.

Reply:
fixed.